src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,70 @@
+import classNames from 'classnames';
+import { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  spinner: boolean;
+  onDeleteTodo: (todoId: number) => void;
+  isDeleting: boolean;
+  deletingCompleted: boolean;
+};
+
+const TodoComponent: React.FC<Props> = ({
+  todo, spinner, onDeleteTodo, isDeleting, deletingCompleted,
+}) => {
+  const [hasSpinner, setHasSpinner] = useState(spinner);
+
+  const isActive = todo.completed && deletingCompleted ? true : hasSpinner;
+
+  useEffect(() => {
+    if (!isDeleting) {
+      setHasSpinner(false);
+    }
+  }, [isDeleting]);
```

Comment:
You don't need this. You can check `isDeleting` directly.

Reply:
Because you handle this statement in `onDeleteTodo` already.

Reply:
I implemented this commentary and the following ones, and that broke my code((( it seems that i do need this deleting completed and cannot do only with isDeleting, as deleted completed handles only the deleting of completed tasks and isDeleting with individual ones. Therefore when i set the spinner, i need to handle the two states separately. there may be different logic but i will stick to this one for now as i will break my brain trying to redo the logic of the whole app))